genotyping/src/org/labkey/genotyping/ImportAnalysisJob.java

```diff
@@ -218,4 +213,20 @@ private QueryContext(DbSchema schema, TableInfo matches, TableInfo reads, int ru
             this.run = run;
         }
     }
+
+    // Simple test that renders mhcQuery.jsp. Useful for validating JspTemplate and non-HTML JSP rendering.
+    public static class TestCase extends Assert
+    {
+        @Test
+        public void testMhcQuery() throws Exception
+        {
+            GenotypingSchema gs = GenotypingSchema.get();
+            DbSchema schema = gs.getSchema();
+            QueryContext ctx = new QueryContext(schema, gs.getMatchesTable(), gs.getReadsTable(), 37);
+            JspTemplate<QueryContext> jspQuery = new JspTemplate<>("/org/labkey/genotyping/view/mhcQuery.jsp", ctx);
+            String sql = jspQuery.render();
+            int expectedLength = gs.getSqlDialect().isSqlServer() ? 1386 : 1405;
```

Comment:
Length seems potentially fragile, but maybe this won't change often

Reply:
Yes, in fact this test is failing on TeamCity, claiming a length of 11728. I'll likely get rid of the length check, but I started outputting the SQL because I'm curious what's causing this discrepancy.